website/identifiers/utils.py

```diff
@@ -142,16 +145,16 @@ def get_or_create_identifiers(target_object):
 
         resp = response_dict['response']
         exists = response_dict['already_exists']
-
+        new = response_dict['new']
         if exists:
             doi = resp['success']
             suffix = doi.strip(settings.DOI_NAMESPACE)
-            return {
-                'doi': doi.replace('doi:', ''),
-                'ark': '{0}{1}'.format(settings.ARK_NAMESPACE.replace('ark:', ''), suffix),
-            }
+            if not new:
+                return {
+                    'doi': doi.replace('doi:', ''),
+                    'ark': '{0}{1}'.format(settings.ARK_NAMESPACE.replace('ark:', ''), suffix),
+                }
+            else:
+                return {'doi': doi.replace('doi:', '')}
         else:
-            return dict(
-                [each.strip('/') for each in pair.strip().split(':')]
-                for pair in resp['success'].split('|')
-            )
+            return {'doi': resp['success'].strip().split(':')[1]}
```

Comment:
I think this if statement can be simplified to not need a nested `if/else` if you use `exists` instead of `new`. Also see above comment about using a specific dict key of `doi` instead of relying on an index